HerPublicWebsite/Middleware/SecurityHeadersMiddleware.cs

```diff
@@ -18,14 +22,19 @@ public Task Invoke(HttpContext context)
         {
             context.Response.Headers.Add("X-Content-Type-Options", "nosniff");
         }
-        
+
+        var nonce = Guid.NewGuid().ToString("N");;
+        context.Items[NonceContextKey] = nonce;
+
         if (!context.Response.Headers.ContainsKey("Content-Security-Policy"))
         {
+            // Based on https://csp.withgoogle.com/docs/strict-csp.html
             context.Response.Headers.Add("Content-Security-Policy",
-                "default-src 'self'; script-src * 'unsafe-inline' 'unsafe-eval'; connect-src  * 'unsafe-inline'"); 
-
+                "object-src 'none'; " +
+                $"script-src 'nonce-{nonce}' 'unsafe-inline' 'strict-dynamic' https:; " +
+                "base-uri 'none';");
```

Comment:
I have removed a couple of bits:
- `'unsafe-eval'` - I don't think we need eval anywhere
- `http:` - I don't think we need to load any resources over HTTP (even if it's only a fallback)